### DISCUSSION TITLE
AspNetCore Authentication 2.2.0 que está deprecado

### DISCUSSION THREAD

**Discussion by AlexandreQuintela** [balta-io/forum#519] [Tira Dúvidas] [answered]
Ola pessoal,

Na aula de token service, o Balta instala o pacote Microsoft AspNetCore Authentication 2.2.0 que está deprecado.
[This package has been deprecated as it is legacy and is no longer maintained. ](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Core/2.2.0?_src=template)
Qual substitui esse pacote?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @AlexandreQuintela, nos cursos novos eu já mostro a atualização...

Os novos pacotes são:
JWT Auth ([Microsoft.AspNetCore.Authentication.JwtBearer](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.JwtBearer))
Facebook OAuth ([Microsoft.AspNetCore.Authentication.Facebook](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Facebook))
Google OAuth ([Microsoft.AspNetCore.Authentication.Google](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Google))
Microsoft Account authentication ([Microsoft.AspNetCore.Authentication.MicrosoftAccount](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.MicrosoftAccount))
OpenID Connect authentication ([Microsoft.AspNetCore.Authentication.OpenIdConnect](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.OpenIdConnect))
OpenID Connect bearer token ([Microsoft.AspNetCore.Authentication.JwtBearer](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.JwtBearer))
Twitter OAuth ([Microsoft.AspNetCore.Authentication.Twitter](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Twitter))
WsFederation authentication ([Microsoft.AspNetCore.Authentication.WsFederation](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.WsFederation))
